client/my-sites/checkout/upsell-nudge/purchase-modal/index.tsx

```diff
@@ -93,12 +98,27 @@ export function wrapValueInManagedValue( value: string | undefined ): ManagedVal
 }
 
 function PurchaseModalWrapper( props: PurchaseModalProps ) {
-	const { onClose, productToAdd, siteSlug, showFeatureList } = props;
+	const {
+		onClose,
+		onPurchaseSuccess = null,
+		disableThankYouPage,
+		productToAdd,
+		siteSlug,
+		showFeatureList,
+	} = props;
 
-	const onComplete = useCreatePaymentCompleteCallback( {
+	const paymentCompleteCallback = useCreatePaymentCompleteCallback( {
 		isComingFromUpsell: true,
 		siteSlug: siteSlug,
+		isInModal: true,
+		disabledThankYouPage: disableThankYouPage,
```

Comment:
Not sure if it's intentional, but one is `disabledThankYouPage` and one is `disableThankYouPage` here. Should they be the same? 

Also, I don't see a case we wouldn't want to disable the thank-you page for the 1-click checkout modal. Unless there is a case, it'd be great to remove it to simplify the component interface :)

Reply:
> Not sure if it's intentional, but one is disabledThankYouPage and one is disableThankYouPage here. Should they be the same?

That's a good spot! My original intention was to have the prop as `disableThankYouPage` as it seems like a "disable?" question whereas `disabledThankYouPage` conveys that it has already been "disabled". But reading your comment, I realize it would probably cause additional confusion, so I've changed the prop to `disabledThankYouPage` for consistency.

> Also, I don't see a case we wouldn't want to disable the thank-you page for the 1-click checkout modal. Unless there is a case, it'd be great to remove it to simplify the component interface :)

In the BBE flow, we need to redirect to the thank you page as it handles the further redirection to the content submission form. :)

The props are starting to increase so I have created a separate PR with documentation here: https://github.com/Automattic/wp-calypso/pull/85614